gpii/node_modules/processReporter/src/processesBridge.js

```diff
@@ -0,0 +1,236 @@
+/*!
+GPII Process Reporter processes bridge -- gpii.processes.
+
+Copyright 2014 Inclusive Design Research Centre, OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+/*global require */
+
+"use strict";
+
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+gpii.processes = fluid.registerNamespace("gpii.processes");
+
+fluid.defaults("gpii.processes", {
+    gradeNames: ["gpii.processes.native", "fluid.component"],
+    invokers: {
+        findSolutionsByCommands: {
+            funcName: "gpii.processes.findSolutionsByCommands",
+            args: ["{arguments}.0"]   // array of command names
+        },
+        findSolutionsByPids: {
+            funcName: "gpii.processes.findSolutionsByPids",
+            args: ["{arguments}.0"]   // array of pids (process ids)
+        },
+        findProcessByPid: {
+            funcName: "gpii.processes.findProcessByPid",
+            args: ["{arguments}.0", "{arguments}.1"] // pid, procArray (optional)
+        },
+        findProcessesByCommand: {
+            funcName: "gpii.processes.findProcessesByCommand",
+            args: ["{arguments}.0", "{arguments}.1"]
+            // command, procArray (optional)
+        },
+        findFirstProcessByCommand: {
+            funcName: "gpii.processes.findFirstProcessByCommand",
+            args: ["{arguments}.0", "{arguments}.1"]
+            // command, procArray (optional)
+        },
+        isRunning: {
+            funcName: "gpii.processes.isRunning",
+            args: ["{arguments}.0"]   // state (string)
+        },
+        updateProcInfo: {
+            funcName: "gpii.processes.updateProcInfo",
+            args: ["{arguments}.0"]   // process info structure
+        },
+        initProcInfoNotRunning: {
+            funcName: "gpii.processes.initProcInfoNotRunning",
+            args: ["{arguments}.0"]   // command name (string)
+        }
+    }
+});
+
+/**
+ * Return a list of process information objects corresponding to the names
+ * of each of the passed in commands.  If nothing is found, returns an empty
+ * array.
+ *
+ * @param commandNames {Array} - the names of the processes to inspect.
+ * @return {Array} - Array of procInfo objects for each command name.  Empty
+ *                   if no processes are found.
+ */
+gpii.processes.findSolutionsByCommands = function (commandNames) {
+    if (!!commandNames) {
```

Comment:
This guard doesn't add much, especially since you don't declare the argument as optional in the docs - I think we can just implement this with the main return statement